### ISSUE TITLE
sslmitm, SNI and proxy issue

### ISSUE THREAD

**Issue by dbrody2004** [e2guardian/e2guardian#576]
Having issue with sslmitm, SNI and proxy.

Configuration is sslmitm = on.  Client issues a web request to IP 35.227.237.213 and adds SNI of www.feelinsonice.com.  e2guardian rewrites the certificate and then proxies to www.feelinsonice.com.  The problem is that when the proxy does a DNS lookup, www.feelinsonice.com doesn't have a DNS entry and so communication fails.

Tried putting fellinsonice.com in both localexceptionsitelist and exceptionsitelist, but it still seems that e2guardian is proxying to www.feelinsonice.com based upon SNI and it doesn't have a DNS entry.

Is there a way around this?

**Comment by Raifeg** [comment]
What is your `originalip` setting in e2guardian.conf? Have you tried it "off"?

**Comment by forid786** [comment]
@dbrody2004 If I remember correctly, this is required to get Snapchat to work?

I managed to get it working by adding it to the exceptions list, I'm running E2G on pfSense. It may work slightly different there to other platforms.

**Comment by dbrody2004** [comment]
I am concerned that setting originalip off will break DNS round robin.

**Comment by philipianpearce** [comment]
Hi, 

Originalip is not compiled by default (and can only be enabled on Linux), and if compiled and switched on is likely to block a lot of sites which use round-robin and/or dynamic DNS so best left off. This option is likely to removed in future releases. 

Philip 

----- Original Message -----

**Comment by philipianpearce** [comment]
Any legitimate SNI will be resolvable in DNS so something odd is happening with whatever is calling this site. 

If you really want to enable it add an entry to /etc/hosts on your proxy:- 

35.227.237.213 www.fellinsonice.com 

Philip 
----- Original Message -----

**Comment by dbrody2004** [comment]
I would agree about SNI legitimacy.  In this case www.feelinsonice.com is part of Snapchat which I would say is a fairly legitimate application.  I also came across this with a siriusxm url as well.  My take is that they are trying to stop transparent proxies that rely on SNI.

In the end we have a script that queries for the real DNS entry in a cronjob and adds it to the host file just in case the IP ever changes.

**Comment by philipianpearce** [comment]
In order to solve this type of issue we need an option to allow the use of the the original destination IP and port instead of looking up SNI in DNS. This would also be more efficient as there would then be no need to make a DNS lookup. 

I will raise a ticket for this enhancement as soon as I am able (I'm away at the moment with very slow and intermittent Internet.) 

I have already started work on this together with a review of the transparent logic for https and https and will have it posted to v5.4.dev for testing by the end of February. 

This work should also solve a number of related issues such as the looping issues experienced on BSD platforms due to the current failure to look up correctly the original destination ip/port on any non-linux platform. 

Once I have posted the enhancement issue I will close this issue. 

Regards 
Philip 

----- Original Message -----
